### PR TITLE
Add gh action template

### PR DESCRIPTION
**Pull request by JamesGlover**
Adds a template for an action which will automatically update Rubocop.

Tools like depfu are insufficient, as Rubocop updates regularly change the rules
resulting in failed lint builds following an update. This is especially problematic
if we attempt to update Gems in bulk.

### CODE REVIEW

File: workflow-templates/update_rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  workflow_dispatch:
+  schedule:
+  # * is a special character in YAML so you have to quote this string
+  # Every sunday at 10 am
+  - cron: "0 10 * * 0"
+
+jobs:
+  update_rubocop:
+    if: (github.repository_owner == 'sanger') # On the sanger fork only
+    runs-on: ubuntu-latest
```
Comment (pjvv) [inline workflow-templates/update_rubocop.yml:30]: Should we pin this to what we use in FCE instead or is it low risk?

File: workflow-templates/update_rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      run: bundle exec rubocop --auto-correct --disable-uncorrectable
+      # https://github.com/marketplace/actions/create-pull-request
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v3
+      with:
+        title: Automatic Rubocop update
+        branch: gh_action/update_rubocop
+        delete-branch: true
+        body: |
+          Automatic update via .github/workflows/update_rubocop.yml
```
Comment (pjvv) [inline workflow-templates/update_rubocop.yml:53]: Full link might be useful here.

**Review by pjvv — APPROVED** [review]
Looks cool, would like to see it in action!

File: workflow-templates/update_rubocop.yml (hunk 1 above)
Comment (JamesGlover) [reply workflow-templates/update_rubocop.yml:30]: For linting alone, I'm going to say pretty low risk.

I haven't been version pinning the test suites, perhaps we should. When I started the process of migrating, they didn't even offer version pinning, or it was poorly documented, that got added as they started the migration from 18.0.4 to 20.04)

That said, it should still be fairly low risk for most the SS stuff, as we're not actually packaging the binaries we generate as part of the build.

File: workflow-templates/update_rubocop.yml (hunk 2 above)
Comment (JamesGlover) [reply workflow-templates/update_rubocop.yml:53]: It adds reasonable complexity to the action, you need something like this:
```
    - name: Generate url
        id: urls
        run: echo ::set-output name=action-url::"$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/blob/$GITHUB_SHA/"
      # https://github.com/marketplace/actions/create-pull-request
    - name: Create Pull Request
      uses: peter-evans/create-pull-request@v3
      with:
        title: Automatic Rubocop update
        branch: gh_action/update_rubocop
        delete-branch: true
        body: Automatic update via ${{ steps.urls.outputs.action-url }}/.github/workflows/update_rubocop.yml
```

**Pull request merged**